packages/mds-metrics/types.ts

```diff
@@ -0,0 +1,48 @@
+import { ApiRequest, ApiResponse } from '@mds-core/mds-api-server'
+import {
+  VEHICLE_STATUS,
+  VEHICLE_EVENT,
+  VEHICLE_TYPES,
+  VEHICLE_TYPE,
+  VEHICLE_EVENTS,
+  VEHICLE_STATUSES
+} from '@mds-core/mds-types'
+
+export type MetricsApiRequest = ApiRequest
+export type MetricsApiResponse = ApiResponse
+
+export type StateSnapshotResponse = {
+  [T in VEHICLE_TYPE]: { [S in VEHICLE_STATUS]: number }
+}
+
+export type EventSnapshotResponse = {
+  [T in VEHICLE_TYPE]: { [S in VEHICLE_EVENT]: number }
+}
+
+export type CountsResponse = {
+  [T in VEHICLE_TYPE]: number
+}
```

Comment:
Could also leverage these types in the API response declarations and request handlers.
```
export type MetricsApiResponse<T> = ApiResponse<T>
export type GetStateSnapshotResponse = MetricsApiResponse<StateSnapshotResponse>
export type GetEventsSnapshotResponse = MetricsApiResponse<EventSnaphotResponse>
export type GetTelemetryCountsResponse = MetricsApiResponse<CountsReponse>
export type GetEventCountsResponse = MetricsApiResponse<CountsResponse>
```